sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2586,6 +2586,15 @@ object SQLConf {
       .checkValue(_ > 0, "The timeout value must be positive")
       .createWithDefault(10L)
 
+  val LEGACY_ALLOW_MISSING_YEAR_DURING_PARSING =
+    buildConf("spark.sql.legacy.allowMissingYearDuringParsing")
+      .internal()
+      .doc("When true, DateFormatter/TimestampFormatter allows parsing date/timestamp string " +
```

Comment:
Here too, you could suggest the alternative workaround that doesn't require setting the legacy flag.